packages/codemods/src/codeMods/utilities/helpers/propHelpers.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  ts,
+  Node,
+  SyntaxKind,
+  JsxOpeningElement,
+  JsxSelfClosingElement,
+  VariableDeclarationKind,
+  CodeBlockWriter,
+} from 'ts-morph';
+import { ValueMap } from 'src/codeMods/types';
+import { Maybe } from '../../../maybe';
+
+/* Helper function to rename a prop if in a spread operator.  */
+export function renamePropInSpread(
+  element: JsxOpeningElement | JsxSelfClosingElement,
+  toRename: string,
+  replacementName: string,
+  changeValueMap?: ValueMap<string>,
+  replacementValue?: string,
+) {
+  const allAttributes = element.getAttributes();
+  allAttributes.forEach(attribute => {
+    if (attribute.getKind() === SyntaxKind.JsxSpreadAttribute) {
+      /* Get the name of the object that houses the prop. */
+      const spreadProp = attribute.getFirstChildByKind(SyntaxKind.Identifier);
+      if (
+        spreadProp &&
+        spreadProp
+          .getType()
+          .getProperties()
+          .some(name => name.getName() === toRename)
+      ) {
+        /* If we've found our prop inside the object,
+          continue attempt to rename it. */
+        const propKind = spreadProp.getDefinitions();
+        if (propKind.length === 1) {
+          const newSpreadName = '__mig' + propKind[0].getName()[0].toUpperCase() + propKind[0].getName().substring(1);
+          const newMapName = '__migEnumMap';
+          switch (propKind[0].getKind()) {
```

Comment:
I know it was like this initially, but I believe that it's time for this switch statement to get put into somethign like a `shouldUpdateElementProp` function that takes in the kind of the prop and then determines whether or not to run it. The same is true for the spreadProp checks in the above if.